### ISSUE TITLE
Help to code Multivariate Time Series Macro Model

### ISSUE THREAD

**Issue by hyperfra** [pyro-ppl/numpyro#976]
Hi
I think I am not understanding well how lax.scan works. As an example I would like to estimate the following model 

![image](https://user-images.githubusercontent.com/39092766/112643966-cfc66600-8e3c-11eb-8982-b368dd3092ce.png)

my attempt to code it is as follows but I think I am messing up with scan. Can u help me?:

```python

def Havelmo(y,future=0):
       
    alpha   = numpyro.sample("alpha", dist.Uniform(0, 3))
    beta    = numpyro.sample("beta", dist.Uniform(0, 3))
    theta1  = numpyro.sample("theta1", dist.Uniform(0, 3))
    theta0  = numpyro.sample("theta0", dist.Uniform(0, 3))
    gamma1  = numpyro.sample("gamma1", dist.Uniform(0, 2))
    gamma0  = numpyro.sample("gamma0", dist.Uniform(0, 2))
    eps_std = numpyro.sample("eps_std", dist.InverseGamma(5))
    eta_std = numpyro.sample("eta_std", dist.InverseGamma(5))
    v_std   = numpyro.sample("v_std", dist.InverseGamma(5))   
        
    Gamma0 = jnp.array([
                      [1-alpha, -alpha, -alpha],
                      [-theta1, 1, 0],
                      [0, 0, 1],
                      ])    
    C = jnp.array([
                [beta],
                [theta0],
                [gamma0],
                ])        
      
    Gamma1 = jnp.array([
                    [0, 0, 0],
                    [-theta1, 0, 0],
                    [0, 0, gamma1],
                    ])       
       
    VV = jnp.array([
                [eps_std**2, 0, 0],
                [0, eta_std**2, 0],
                [0, 0, v_std**2],
                ])
       
    def transition_fn(carry, t):
        xt = carry
        G = jnp.linalg.inv(Gamma0)
        MU = jnp.matmul(G,C) + jnp.expand_dims(jnp.matmul(G,jnp.matmul(Gamma1,xt)),axis=1)
        MU  = MU.flatten()
        V  = jnp.matmul(jnp.matmul(G,VV),G.T)
        Y_ = numpyro.sample("Y", dist.MultivariateNormal(MU, V))
        xt = Y_
        return (xt), Y_

    N = y.shape[1]
    xt_init = y[:,0]
        
    with numpyro.handlers.condition(data={"Y": y[:,1:]}):
        _, ys = scan(
             transition_fn,  (xt_init), (jnp.arange(1, N))
        )
    
kernel = NUTS(Havelmo)
mcmc = MCMC(kernel, num_warmup=1000, num_samples=2000, num_chains=4)
mcmc.run(random.PRNGKey(0), endog)
mcmc.print_summary()
samples = mcmc.get_samples()

´´´

**Comment by fehiepsi** [comment]
Just some suggestions:
+ set `data={"Y": y[:, 1:].T}` if the last dimension of `y` is time dimension
+ check if the first item of the output of `transition_fn(xt_init, 0)` has the same shape as `xt_init`

**Comment by hyperfra** [comment]
Thanks 🙏


Sent from my iPhone

On 26 Mar 2021, at 17:49, Du Phan ***@***.***> wrote:

﻿

Just some suggestions:

  *   set data={"Y": y[:, 1:].T} if the last dimension of y is time dimension
  *   check if the first item of the output of transition_fn(xt_init, 0) has the same shape as xt_init

**Comment by fehiepsi** [comment]
@hyperfra Did you make this work? I think transposing `y` will resolve the issue for you. Please follow-up using either [forum](https://forum.pyro.ai/) or comments below.

**Comment by hyperfra** [comment]
hi I did. you were right just transposing made it work. thank you again
